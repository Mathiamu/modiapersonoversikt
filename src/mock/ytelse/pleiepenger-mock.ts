import * as faker from 'faker/locale/nb_NO';
import * as moment from 'moment';

import navfaker from 'nav-faker/dist/index';
import {
    Arbeidsforhold, Periode,
    Pleiepengeperiode,
    Pleiepengerettighet,
    PleiepengerResponse, Vedtak
} from '../../models/ytelse/pleiepenger';
import { fyllRandomListe } from '../utils/mock-utils';

export function getPleiepenger(fødselsnummer: string): PleiepengerResponse {
    faker.seed(Number(fødselsnummer));
    navfaker.seed(fødselsnummer);

    return {
        pleiepenger: fyllRandomListe(() => getPleiepengerettighet(fødselsnummer), 10)
    };
}

function getPleiepengerettighet(fødselsnummer: string): Pleiepengerettighet {
    return {
        barnet: navfaker.fødselsnummer.generer(),
        omsorgsperson: fødselsnummer,
        andreOmsorgsperson: navfaker.fødselsnummer.generer(),
        restDagerFomIMorgen: navfaker.random.number(100),
        forbrukteDagerTomIDag: navfaker.random.number(20),
        pleiepengedager: navfaker.random.number(200),
        restDagerAnvist: navfaker.random.number(50),
        perioder: fyllRandomListe(() => getPleiepengeperiode(), 10)
    };
}

function getPleiepengeperiode(): Pleiepengeperiode {
    return {
        fom: moment(faker.date.recent()).format(moment.ISO_8601.__momentBuiltinFormatBrand),
        antallPleiepengedager: navfaker.random.number(100),
        arbeidsforhold: fyllRandomListe(() => getArbeidsforhold(), 10),
        vedtak: fyllRandomListe(() => getVedtak(), 10)
    };
}

function getArbeidsforhold(): Arbeidsforhold {
    return {
        arbeidsgiverNavn: faker.company.companyName(),
        arbeidsgiverKontonr: Number(faker.finance.account(9)).toString(),
        inntektsperiode: moment(faker.date.recent()).format(moment.ISO_8601.__momentBuiltinFormatBrand),
        inntektForPerioden: Number(faker.commerce.price()),
        refusjonTom: moment(faker.date.recent()).format(moment.ISO_8601.__momentBuiltinFormatBrand),
        refusjonstype: 'REFUSJONTYPE',
        arbeidsgiverOrgnr: '1234567890',
        arbeidskategori: 'ARBKAT'
    };
}

function getVedtak(): Vedtak {
    return {
        periode: getPeriode(),
        kompensasjonsgrad: navfaker.random.number(90),
        utbetalingsgrad: navfaker.random.number(90),
        anvistUtbetaling: 'ANVIST',
        bruttobeløp: Number(faker.commerce.price()),
        dagsats: navfaker.random.number(70),
        pleiepengegrad: navfaker.random.number(60)
    };
}

function getPeriode(): Periode {
    return {
        fom: moment(faker.date.recent()).format(moment.ISO_8601.__momentBuiltinFormatBrand),
        tom: moment(faker.date.recent()).format(moment.ISO_8601.__momentBuiltinFormatBrand)
    };
}