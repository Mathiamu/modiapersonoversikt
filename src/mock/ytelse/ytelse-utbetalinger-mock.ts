import * as moment from 'moment';

import {
    HistoriskUtbetaling,
    KommendeUtbetaling,
    KreditorTrekk,
    UtbetalingPåVent
} from '../../models/ytelse/ytelse-utbetalinger';
import { getPeriode } from '../person/periodeMock';
import FakerStatic = Faker.FakerStatic;
import { fyllRandomListe } from '../utils/mock-utils';

export function getHistoriskUtbetaling(faker: FakerStatic): HistoriskUtbetaling {
    return {
        vedtak: getPeriode(),
        utbetalingsgrad: faker.random.number(100),
        utbetalingsdato: moment(faker.date.recent()).format(moment.ISO_8601.__momentBuiltinFormatBrand),
        nettobeløp: Number(faker.commerce.price()),
        bruttobeløp: Number(faker.commerce.price()),
        skattetrekk: Number(faker.random.number(50)),
        arbeidsgiverNavn: faker.company.companyName(),
        arbeidsgiverOrgNr: '1234567890',
        dagsats: Number(faker.commerce.price()),
        type: 'KONTOØVERFØRING',
        trekk: fyllRandomListe(() => getKreditortrekk(faker), 3)
    };
}

export function getKommendeUtbetaling(faker: FakerStatic): KommendeUtbetaling {
    return {
        vedtak: getPeriode(),
        utbetalingsgrad: faker.random.number(100),
        utbetalingsdato: moment(faker.date.recent()).format(moment.ISO_8601.__momentBuiltinFormatBrand),
        bruttobeløp: Number(faker.commerce.price()),
        arbeidsgiverNavn: faker.company.companyName(),
        arbeidsgiverOrgNr: '1234567890',
        arbeidsgiverKontonr: Number(faker.finance.account(9)).toString(),
        dagsats: Number(faker.commerce.price()),
        saksbehandler: faker.name.firstName() + ' ' + faker.name.lastName(),
        type: 'KONTOØVERFØRING'
    };
}

export function getUtbetalingPåVent(faker: FakerStatic): UtbetalingPåVent {
    return {
        vedtak: getPeriode(),
        utbetalingsgrad: faker.random.number(100),
        oppgjørstype: 'Oppgjørstype',
        arbeidskategori: 'Arbeidskategori',
        stansårsak: 'Stansårsak',
        ferie1: getPeriode(),
        ferie2: getPeriode(),
        sanksjon: getPeriode(),
        sykmeldt: getPeriode()
    };
}

function getKreditortrekk(faker: FakerStatic): KreditorTrekk {
    return {
        kreditorsNavn: faker.company.companyName(),
        beløp: Number(faker.commerce.price())
    };
}