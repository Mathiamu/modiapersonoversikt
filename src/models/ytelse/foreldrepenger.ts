import { Periode } from '../periode';
import { HistoriskUtbetaling, KommendeUtbetaling } from './ytelse-utbetalinger';

export interface ForeldrepengerResponse {
    foreldrepenger?: Foreldrepengerrettighet;
}

export interface Foreldrepengerrettighet {
    forelder: string;
    andreForeldersFnr?: string;
    antallBarn?: number;
    barnetsFødselsdato?: string;
    dekningsgrad: number;
    fedrekvoteTom?: string;
    mødrekvoteTom?: string;
    foreldrepengetype?: string;
    graderingsdager?: number;
    restDager: number;
    rettighetFom?: string;
    eldsteIdDato?: string;
    foreldreAvSammeKjønn?: string;
    periode?: Foreldrepengerperiode[];
}

export interface Foreldrepengerperiode {
    fødselsnummer: string;
    harAleneomsorgFar?: boolean;
    harAleneomsorgMor?: boolean;
    arbeidsprosentMor?: number;
    avslagsårsak?: string;
    avslått?: string;
    disponibelGradering?: number;
    erFedrekvote?: boolean;
    erMødrekvote?: boolean;
    forskyvelsesårsak1?: string;
    forskyvelsesperiode1?: Periode;
    forskyvelsesårsak2?: string;
    forskyvelsesperiode2?: Periode;
    foreldrepengerFom: string;
    midlertidigStansDato?: string;
    morSituasjon?: string;
    rettTilFedrekvote: string;
    rettTilMødrekvote: string;
    stansårsak?: string;
    historiskeUtbetalinger?: HistoriskUtbetaling[];
    kommendeUtbetalinger?: KommendeUtbetaling[];
}