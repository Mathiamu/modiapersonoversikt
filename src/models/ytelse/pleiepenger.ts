export interface PleiepengerResponse {
    pleiepenger: Pleiepengerettighet[] | null;
}

export interface Pleiepengerettighet {
    barnet: string;
    omsorgsperson: string;
    andreOmsorgsperson?: string;
    restDagerFomIMorgen: number;
    forbrukteDagerTomIDag: number;
    pleiepengedager: number;
    restDagerAnvist: number;
    perioder?: Pleiepengeperiode[];
}

export interface Pleiepengeperiode {
    fom: string;
    antallPleiepengedager: number;
    arbeidsforhold?: Arbeidsforhold[];
    vedtak?: Vedtak[];
}

export interface Arbeidsforhold {
    arbeidsgiverNavn: string;
    arbeidsgiverKontonr?: string;
    inntektsperiode?: string;
    inntektForPerioden?: number;
    refusjonTom?: string;
    refusjonstype?: string;
    arbeidsgiverOrgnr: string;
    arbeidskategori: string;
}

export interface Vedtak {
    periode: Periode;
    kompensasjonsgrad?: number;
    utbetalingsgrad: number;
    anvistUtbetaling: string;
    bruttobel√∏p: number;
    dagsats: number;
    pleiepengegrad?: number;
}

export interface Periode {
    fom: string;
    tom: string;
}