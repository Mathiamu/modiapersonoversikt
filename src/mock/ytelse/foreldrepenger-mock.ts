import * as faker from 'faker/locale/nb_NO';
import * as moment from 'moment';

import navfaker from 'nav-faker/dist/index';

import {
    Foreldrepengerperiode,
    ForeldrepengerResponse,
    Foreldrepengerrettighet
} from '../../models/ytelse/foreldrepenger';
import { getPeriode } from '../person/periodeMock';
import { fyllRandomListe } from '../utils/mock-utils';
import { getHistoriskUtbetaling, getKommendeUtbetaling } from './ytelse-utbetalinger-mock';
import { HistoriskUtbetaling, KommendeUtbetaling } from '../../models/ytelse/ytelse-utbetalinger';

export function getMockForeldrepenger(fødselsnummer: string): ForeldrepengerResponse {
    faker.seed(Number(fødselsnummer));
    navfaker.seed(fødselsnummer + 'foreldrepenger');

    if (navfaker.random.vektetSjanse(0.3)) {
        return {};
    }

    return {
        foreldrepenger: getForeldrepengerrettighet(fødselsnummer)
    };
}

function getForeldrepengerrettighet(fødselsnummer: string): Foreldrepengerrettighet {
    return {
        forelder: fødselsnummer,
        andreForeldersFnr: navfaker.personIdentifikator.fødselsnummer(),
        antallBarn: navfaker.random.integer(5),
        barnetsFødselsdato: moment(faker.date.recent()).format(moment.ISO_8601.__momentBuiltinFormatBrand),
        dekningsgrad: navfaker.random.integer(95),
        fedrekvoteTom: moment(faker.date.recent()).format(moment.ISO_8601.__momentBuiltinFormatBrand),
        mødrekvoteTom: moment(faker.date.recent()).format(moment.ISO_8601.__momentBuiltinFormatBrand),
        foreldrepengetype: 'FTYPE',
        graderingsdager: navfaker.random.integer(100),
        restDager: navfaker.random.integer(50),
        rettighetFom: moment(faker.date.recent()).format(moment.ISO_8601.__momentBuiltinFormatBrand),
        eldsteIdDato: moment(faker.date.recent()).format(moment.ISO_8601.__momentBuiltinFormatBrand),
        periode: fyllRandomListe<Foreldrepengerperiode>(() => getForeldrepengerperiode(fødselsnummer), 10)
    };
}

function getForeldrepengerperiode(fødselsnummer: string): Foreldrepengerperiode {
    return {
        fødselsnummer: fødselsnummer,
        harAleneomsorgFar: faker.random.boolean(),
        harAleneomsorgMor: faker.random.boolean(),
        arbeidsprosentMor: navfaker.random.integer(100),
        avslagsårsak: 'AVSLAG',
        avslått: 'AVSLÅTT',
        disponibelGradering: navfaker.random.integer(75),
        erFedrekvote: faker.random.boolean(),
        erMødrekvote: faker.random.boolean(),
        forskyvelsesårsak1: 'FÅRSAK1',
        forskyvelsesperiode1: getPeriode(),
        forskyvelsesårsak2: 'FÅRSAK2',
        forskyvelsesperiode2: getPeriode(),
        foreldrepengerFom: moment(faker.date.recent()).format(moment.ISO_8601.__momentBuiltinFormatBrand),
        midlertidigStansDato: moment(faker.date.recent()).format(moment.ISO_8601.__momentBuiltinFormatBrand),
        morSituasjon: faker.lorem.words(5),
        rettTilFedrekvote: 'RETTF',
        rettTilMødrekvote: 'RETTM',
        stansårsak: 'STANS',
        historiskeUtbetalinger: fyllRandomListe<HistoriskUtbetaling>(() => getHistoriskUtbetaling(faker), 5),
        kommendeUtbetalinger: fyllRandomListe<KommendeUtbetaling>(() => getKommendeUtbetaling(faker), 5)
    };
}