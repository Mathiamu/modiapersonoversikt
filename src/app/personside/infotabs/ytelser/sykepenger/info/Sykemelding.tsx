import * as React from 'react';
import { Periode } from '../../../../../../models/periode';
import { getSykemeldingPeriode } from '../utils/sykepengerUtils';
import DescriptionList, { DescriptionListEntries } from '../../../../../../components/DescriptionList';
import { datoEllerNull, periodeEllerNull, prosentEllerNull } from '../../../../../../utils/stringFormatting';
import YtelserInfoGruppe from '../../felles-styling/YtelserInfoGruppe';
import { Sykmelding as ISykemelding } from '../../../../../../models/ytelse/sykepenger';
import styled from 'styled-components';
import theme from '../../../../../../styles/personOversiktTheme';

interface Props {
    sykmeldinger: ISykemelding[];
}

const ListeElementStyle = styled.li`
    ${theme.gr√•ttPanel};
    padding: ${theme.margin.px20};
`;

function GraderingsListeElement({ sykmelding }: { sykmelding: ISykemelding }) {
    const entries: DescriptionListEntries = {
        Periode: periodeEllerNull(sykmelding.sykmeldt),
        Gradering: prosentEllerNull(sykmelding.sykmeldingsgrad)
    };
    return (
        <ListeElementStyle>
            <DescriptionList entries={entries} />
        </ListeElementStyle>
    );
}

function GraderingsListe(props: Props) {
    return (
        <ol>
            {props.sykmeldinger.map((sykmelding, index) => (
                <GraderingsListeElement key={index} sykmelding={sykmelding} />
            ))}
        </ol>
    );
}

function Sykemelding(props: Props) {
    const sykemeldingPeriode: Periode = getSykemeldingPeriode(props.sykmeldinger);
    const aktuellSykemelding = props.sykmeldinger[0];
    const sykemeldingEntries: DescriptionListEntries = {
        Periode: periodeEllerNull(sykemeldingPeriode),
        Behandlingsdato: datoEllerNull(aktuellSykemelding.behandlet),
        Sykmelder: aktuellSykemelding.sykmelder,
        Gradering: <GraderingsListe {...props} />
    };

    return (
        <YtelserInfoGruppe tittel="Sykemelding">
            <DescriptionList entries={sykemeldingEntries} />
        </YtelserInfoGruppe>
    );
}

export default Sykemelding;
