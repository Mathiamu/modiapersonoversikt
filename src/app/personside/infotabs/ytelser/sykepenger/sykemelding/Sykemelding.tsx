import * as React from 'react';
import { Periode } from '../../../../../../models/periode';
import { getSykemeldingPeriode } from '../utils/sykepengerUtils';
import DescriptionList, { DescriptionListEntries } from '../../../../../../components/DescriptionList';
import { datoEllerNull, periodeEllerNull, prosentEllerNull } from '../../../../../../utils/stringFormatting';
import YtelserInfoGruppe from '../../felles-styling/YtelserInfoGruppe';
import { Sykmelding as ISykemelding } from '../../../../../../models/ytelse/sykepenger';
import styled from 'styled-components';
import theme from '../../../../../../styles/personOversiktTheme';
import { createTable } from '../../../../../../utils/tableUtils';

interface Props {
    sykmeldinger: ISykemelding[];
}

const GraderingsTableStyle = styled.div`
    ${theme.table}
`;

function GraderingsTabell(props: Props) {
    const tittelRekke = ['Periode', 'Gradering'];
    const tableEntries = props.sykmeldinger.map(sykmelding => [
        periodeEllerNull(sykmelding.sykmeldt) || undefined,
        prosentEllerNull(sykmelding.sykmeldingsgrad) || undefined
    ]);
    return (
        <GraderingsTableStyle aria-label="Sykmeldingsperioder">
            {createTable(tittelRekke, tableEntries)}
        </GraderingsTableStyle>
    );
}

function Sykemelding(props: Props) {
    const sykemeldingPeriode: Periode = getSykemeldingPeriode(props.sykmeldinger);
    const aktuellSykemelding = props.sykmeldinger[0];
    const sykemeldingEntries: DescriptionListEntries = {
        Periode: periodeEllerNull(sykemeldingPeriode),
        Behandlingsdato: datoEllerNull(aktuellSykemelding.behandlet),
        Sykmelder: aktuellSykemelding.sykmelder
    };

    return (
        <YtelserInfoGruppe tittel="Sykmelding">
            <DescriptionList entries={sykemeldingEntries} />
            <GraderingsTabell {...props} />
        </YtelserInfoGruppe>
    );
}

export default Sykemelding;
