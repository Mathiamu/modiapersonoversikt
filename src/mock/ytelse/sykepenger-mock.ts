import faker from 'faker/locale/nb_NO';
import moment from 'moment';

import navfaker from 'nav-faker/dist/index';

import {
    Forsikring,
    SykepengerResponse,
    Sykmelding,
    Sykepenger,
    Yrkesskade,
    Gradering
} from '../../models/ytelse/sykepenger';
import { getPeriode } from '../person/periodeMock';
import { backendDatoformat, fyllRandomListe } from '../utils/mock-utils';
import { getHistoriskUtbetaling, getKommendeUtbetaling, getUtbetalingPåVent } from './ytelse-utbetalinger-mock';
import { HistoriskUtbetaling, KommendeUtbetaling, UtbetalingPåVent } from '../../models/ytelse/ytelse-utbetalinger';
import { aremark } from '../person/aremark';
import { Arbeidsforhold } from '../../models/ytelse/arbeidsforhold';
import { statiskSykepengerMock } from './statiskSykepengerMock';

export function getMockSykepengerRespons(fødselsnummer: string): SykepengerResponse {
    faker.seed(Number(fødselsnummer));
    navfaker.seed(fødselsnummer + 'sykepenger');

    if (fødselsnummer === aremark.fødselsnummer) {
        return {
            sykepenger: [statiskSykepengerMock, statiskSykepengerMock]
        };
    }

    if (navfaker.random.vektetSjanse(0.3)) {
        return {
            sykepenger: null
        };
    }

    return {
        sykepenger: fyllRandomListe<Sykepenger>(() => getMockSykmepenger(fødselsnummer), 3)
    };
}

export function getMockSykmepenger(fødselsnummer: string): Sykepenger {
    return {
        fødselsnummer: fødselsnummer,
        sykmeldtFom: moment(faker.date.past(1)).format(backendDatoformat),
        forbrukteDager: navfaker.random.integer(100),
        ferie1: navfaker.random.vektetSjanse(0.3) ? getPeriode() : null,
        ferie2: navfaker.random.vektetSjanse(0.3) ? getPeriode() : null,
        sanksjon: navfaker.random.vektetSjanse(0.3) ? getPeriode() : null,
        stansårsak: navfaker.random.vektetSjanse(0.3) ? 'Svindel og bedrag' : null,
        unntakAktivitet: navfaker.random.vektetSjanse(0.3) ? 'Untatt aktivitet' : null,
        forsikring: navfaker.random.vektetSjanse(0.3) ? getForsikring() : null,
        sykmeldinger: fyllRandomListe<Sykmelding>(() => getMockSykmelding(), 3),
        historiskeUtbetalinger: fyllRandomListe<HistoriskUtbetaling>(() => getHistoriskUtbetaling(faker), 5, true),
        kommendeUtbetalinger: fyllRandomListe<KommendeUtbetaling>(() => getKommendeUtbetaling(faker), 3, true),
        utbetalingerPåVent: fyllRandomListe<UtbetalingPåVent>(() => getUtbetalingPåVent(faker), 2, true),
        bruker: fødselsnummer,
        midlertidigStanset: navfaker.random.vektetSjanse(0.3)
            ? moment(faker.date.past(1)).format(backendDatoformat)
            : null,
        slutt: navfaker.random.vektetSjanse(0.7) ? null : moment(faker.date.past(1)).format(backendDatoformat),
        arbeidsforholdListe: fyllRandomListe(() => getArbeidsforhold(), 10, true),
        erArbeidsgiverperiode: navfaker.random.vektetSjanse(0.5),
        arbeidsKategori: 'Ærlig arbeid'
    };
}

function getForsikring(): Forsikring {
    return {
        forsikringsordning: faker.lorem.words(1),
        premiegrunnlag: Number(faker.commerce.price()),
        erGyldig: faker.random.boolean(),
        forsikret: navfaker.random.vektetSjanse(0.5) ? getPeriode() : null
    };
}

export function getMockSykmelding(): Sykmelding {
    return {
        sykmelder: faker.name.firstName() + ' ' + faker.name.lastName(),
        behandlet: moment(faker.date.past(1)).format(backendDatoformat),
        sykmeldt: getPeriode(),
        sykmeldingsgrad: navfaker.random.integer(100),
        gjelderYrkesskade: navfaker.random.vektetSjanse(0.0) ? getYrkesskade() : null,
        gradAvSykmeldingListe: fyllRandomListe(getGradering, 3)
    };
}

function getYrkesskade(): Yrkesskade {
    return {
        yrkesskadeart: faker.lorem.words(3),
        skadet: moment(faker.date.past(1)).format(backendDatoformat),
        vedtatt: moment(faker.date.past(1)).format(backendDatoformat)
    };
}

function getGradering(): Gradering {
    return {
        gradert: getPeriode(),
        sykmeldingsgrad: navfaker.random.integer(100)
    };
}

function getArbeidsforhold(): Arbeidsforhold {
    return {
        arbeidsgiverNavn: faker.company.companyName(),
        arbeidsgiverKontonr: Number(faker.finance.account(11)).toString(),
        inntektsperiode: 'Månedssats',
        inntektForPerioden: Math.round(Number(faker.finance.amount(5000, 50000))),
        refusjonTom: moment(faker.date.past(2)).format(backendDatoformat),
        refusjonstype: 'Ikke refusjon',
        sykepengerFom: moment(faker.date.past(2)).format(backendDatoformat)
    };
}
