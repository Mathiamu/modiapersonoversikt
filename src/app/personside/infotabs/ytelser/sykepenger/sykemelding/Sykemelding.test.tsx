import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { statiskSykepengerMock } from '../../../../../../mock/ytelse/statiskSykepengerMock';
import Sykemelding from './Sykemelding';

test('Sykemelding matcher snapshot', () => {
    const resultat = renderer.create(<Sykemelding sykmeldinger={statiskSykepengerMock.sykmeldinger} />);

    expect(resultat.toJSON()).toMatchSnapshot();
});
