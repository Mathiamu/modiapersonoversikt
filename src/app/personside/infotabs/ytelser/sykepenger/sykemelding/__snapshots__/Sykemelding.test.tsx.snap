// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sykemelding matcher snapshot 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c1 dt {
  color: #78706a;
}

.c1 dd {
  font-weight: bold;
  margin-top: 0.3rem;
}

.c1 > div {
  margin: 0.9375rem 0;
  padding-right: 1rem;
  width: 13rem;
  overflow-wrap: break-word;
}

.c0 {
  border-radius: 0.5rem;
  background-color: #e9e7e7;
  box-shadow: inset 0 0 0 0.3rem white,inset 0 0 0 0.35rem rgba(0,0,0,0.1);
  padding: 1.25rem;
}

.c2 {
  overflow: auto;
  box-shadow: 0 0 0 0.05rem rgba(0,0,0,0.25);
  border-radius: .25rem;
  background-color: white;
}

.c2 table {
  text-align: left;
  width: 100%;
}

.c2 table thead {
  border-bottom: 0.2rem solid rgba(0,0,0,0.15);
  font-weight: bold;
}

.c2 table th,
.c2 table td {
  padding: 0.7rem;
  vertical-align: middle;
}

.c2 table th {
  padding-bottom: 0.7rem;
}

.c2 table tr:nth-child(even) {
  background-color: rgba(0,0,0,0.1);
}

<div
  className="c0"
>
  <h3
    className="typo-ingress"
  >
    Sykmelding
  </h3>
  <dl
    className="c1"
  >
    <div>
      <dt
        className="typo-etikett-liten"
      >
        Periode
      </dt>
      <dd
        className="typo-normal"
      >
        09.02.2014 - 11.05.2014
      </dd>
    </div>
    <div>
      <dt
        className="typo-etikett-liten"
      >
        Behandlingsdato
      </dt>
      <dd
        className="typo-normal"
      >
        01.04.2018
      </dd>
    </div>
    <div>
      <dt
        className="typo-etikett-liten"
      >
        Sykmelder
      </dt>
      <dd
        className="typo-normal"
      >
        Lommelegen AS
      </dd>
    </div>
  </dl>
  <div
    className="typo-normal c2"
  >
    <table
      role="table"
    >
      <thead
        role="rowgroup"
      >
        <tr
          role="row"
        >
          <th
            role="columnheader"
          >
            Periode
          </th>
          <th
            role="columnheader"
          >
            Gradering
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            28.07.2014 - 03.01.2014
          </td>
          <td
            role="cell"
          >
            89%
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
